langchain/agents/agent.py

```diff
@@ -616,14 +654,124 @@ async def _arun(
         return query
 
 
+class ReflexionOutputParser(BaseOutputParser):
+    @abstractmethod
+    def parse(self, text: str) -> str:
+        """Parse text into agent action/finish."""
+
+
+class Reflector(BaseModel):
+    """Agent for the Reflexer chain."""
+
+    llm_chain: LLMChain
+    output_parser: ReflexionOutputParser
+
+    max_iterations_per_trial: Optional[int] = 15
+    max_execution_time_per_trial: Optional[float] = None
+
+    trial_history: List[str] = []
+    """Full string representation of each trial"""
+    trial_reflexions: List[str] = []
+    """Reflexion for of each trial"""
+    trial_prefix: str = "\nTrial {trial_number}"
+    trial_suffix: str = "\nSTATUS: FAIL\nNew plan: "
+
+    def reset(self) -> None:
+        self.trial_history = []
+        self.trial_reflexions = []
+
+    def current_trial_prefix(self, trial_number: int) -> str:
+        return self.trial_prefix.replace("{trial_number}", str(trial_number))
```

Comment:
in MRKL agent, the prompt is formatted using format, e.g. here it would be :
`self.trial_prefix.format(trial_number = str(trial_number))`

From my understanding, the difference with this alternative is that it would raise an error if {trial_number} is not present in self.trial_prefix (while the "replace" implementation would just silently do nothing). Depending on the behavior that you want in this edge case, you may stick to the current implementation (with replace) or the one I provided here (with format) ;-)

Reply:
Thanks! Changed. Went with `format`